src/biotite/sequence/io/fasta/convert.py

```diff
@@ -199,26 +208,41 @@ def set_alignment(fasta_file, alignment, seq_names):
         fasta_file[seq_names[i]] = gapped_seq_strings[i]
 
 
-def _convert_to_sequence(seq_str):
+def _convert_to_sequence(seq_str, seq_type=None):
+
+    # Define preprocessing of preimplemented sequence types
+    process_protein_sequence = lambda x : x.upper().replace("U", "C")
+    process_nucleotide_sequence = (
+        lambda x : x.upper().replace("U","T").replace("X","N")
+    )
+
+    # Attempt to set manually selected sequence type
+    if seq_type is not None:
+        # Do preprocessing as done without manual selection
+        if seq_type == NucleotideSequence:
+            seq_str = process_nucleotide_sequence(seq_str)
+        elif seq_type == ProteinSequence:
+            if "U" in seq_str:
+                warnings.warn(
+                    "ProteinSequence objects do not support selenocysteine "
+                    "(U), occurrences were substituted by cysteine (C)"
+                )
+            seq_str = process_protein_sequence(seq_str)
+        # Return the converted sequence
+        return seq_type(seq_str)    
+
     # Biotite alphabets for nucleotide and proteins
```

Comment:
This part of the comment is also obsolete with the code changes.

```suggestion
```

Reply:
I went over the comments and removed all remainders of the previous implementation. I also added/moved a comment describing the purpose of each lambda function.